test/MathExpressionTests.cpp

```diff
@@ -0,0 +1,91 @@
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include <string>
+#include <set>
+#include "MathExpression.h"
+
+using namespace testing;
+
+TEST(MathExpressionTests, WhenPassedAnEquationWithOneVariable_ThenReturnsAOneItemVectorConatainingThatVariable)
+{
+    std::string equation1 = "foo";
+    std::string equation2 = "bar";
+
+    std::set<std::string> expected1 { "foo" };
+    std::set<std::string> expected2 { "bar" };
+
+    MathExpression mathExpression1(equation1);
+    MathExpression mathExpression2(equation2);
+
+    EXPECT_EQ(expected1, mathExpression1.needsVariableValues());
+    EXPECT_EQ(expected2, mathExpression2.needsVariableValues());
+};
+
+
+
+TEST(MathExpressionTests, WhenPassedAnEquationWithTwoVariables_ThenReturnsATwoItemVectorConatainingThoseVariables)
+{
+    std::string equation1 = "foo+fooo";
+    std::string equation2 = "bar+barr";
+
+    std::set<std::string> expected1 { "foo", "fooo" };
+    std::set<std::string> expected2 { "bar", "barr" };
+
+    MathExpression mathExpression1(equation1);
+    MathExpression mathExpression2(equation2);
+
+    EXPECT_EQ(expected1, mathExpression1.needsVariableValues());
+    EXPECT_EQ(expected2, mathExpression2.needsVariableValues());
+};
+
+
+
+TEST(MathExpressionTests, WhenPassedAnEquationWithTwoVariablesAndOneGetsAValue_ThenReturnsAOneItemVectorConatainingTheUnsetVariable)
+{
+    std::string equation1 = "foo+fooo";
+    std::string equation2 = "bar+barr";
+
+    std::set<std::string> expected1 { "fooo" };
+    std::set<std::string> expected2 { "barr" };
+
+    MathExpression mathExpression1(equation1);
+    MathExpression mathExpression2(equation2);
+
+    mathExpression1.setVariableValue("foo", 1);
+    mathExpression2.setVariableValue("bar", 2);
+
+    EXPECT_EQ(expected1, mathExpression1.needsVariableValues());
+    EXPECT_EQ(expected2, mathExpression2.needsVariableValues());
+};
+
+
+
+TEST(MathExpressionTests, WhenPopulatingATwoVariableString_ThenThePopulatedStringHasReplacedValues)
```

Comment:
Glad to see the new MathExpression tests. Testing is tedious, but it's important for finding out whether your new code works right or not.